book/src/configuration.md

```diff
@@ -255,3 +255,55 @@ render = true
 character = "╎" # Some characters that work well: "▏", "┆", "┊", "⸽"
 skip-levels = 1
 ```
+
+### `[editor.gutters]` Section
+
+For simplicity, `editor.gutters` accepts an array of gutter types, which will
+use default settings for all gutter components.
+
+```toml
+[editor]
+gutters = ["diff", "diagnostics", "line-numbers", "spacer"]
+```
+
+To customize the behavior of gutters, the `[editor.gutters]` section must
+be used. This section contains top level settings, as well as settings for
+specific gutter components as sub-sections.
+
+| Key      | Description                    | Default                                                       |
+| ---      | ---                            | ---                                                           |
+| `layout` | A vector of gutters to display | `["diagnostics", "spacer", "line-numbers", "spacer", "diff"]` |
+
+Example:
+
+```toml
+[editor.gutters]
+layout = ["diff", "diagnostics", "line-numbers", "spacer"]
+```
+
+#### `[editor.gutters.line-numbers]` Section
+
+Options for the line number gutter
+
+| Key         | Description                             | Default |
+| ---         | ---                                     | ---     |
+| `min-width` | The minimum number of characters to use | `3`     |
+
+Example:
+
+```toml
+[editor.gutters.line-numbers]
+min-width = 1
+```
+
+#### `[editor.gutters.diagnotics]` Section
+
+Currently unused
+
+#### `[editor.gutters.diff]` Section
+
+Currently unused
+
+#### `[editor.gutters.spacer]` Section
+
+Currently unused
```

Comment:
What about `[editor.gutters.breakpoints]`? I’m mentioning this because of what I’m doing in #5371 (I might eventually rebase on your work).

Reply:
Right now the sub-sections are styled after the names that can be provided to `editor.gutters`. There's no real reason it _must_ be like this, but it think there's some discoverability benefit to being consistent here. 

Following that style, both breakpoints and diagnostics would be configured in `editors.gutters.diagnostics`. I think it's best to follow this precedent, but I'd say this is a call for some of the more core contributors like @the-mikedavis 

--- 

Looking into the future a bit further, I think the underlying issue here is that diagnostics and breakpoints are a bit oddly coupled - likely a solution from before gutters were configurable. I think there's a case for decoupling diagnostics and breakpoints and trying to tackle that use case more generically. I think it would be interesting to be able to layer gutters as part of the layout. Perhaps something like:

```toml
[editor.gutters]
layout = [ [ "diagnostics", "breakpoints" ], "line-numbers" ]
```

Which would provide the current behavior. I think that needs more discussion before jumping to something like that, though.

Reply:
I’m reimplementing #5371 rebased atop your PR then.